src/main/java/com/google/sps/dao/AssignmentDao.java

```diff
@@ -89,4 +94,112 @@ public ValidationResponse createAssignment(Assignment assignment) {
         }
         return assignments;
     }
+
+    @Override
+    @SneakyThrows
+    public ValidationResponse updateAssignmentForStudent(Assignment newAssignment) {
+
+        ValidationResponse.Builder validationResponseBuilder = null;
+
+        Assignment oldAssignment = getAssignmentById(newAssignment.getId());
+        newAssignment.setId(oldAssignment.getId());
+        newAssignment.setUpdated(System.currentTimeMillis());
+
+        if (ValidationUtils.isEmptyOrWhiteSpace(newAssignment.getSolution())) {
```

Comment:
Let's do the validations first before we get the old assignment i.e. right after Line 102.

Reply:
done